packages/twoside/init.lua

```diff
@@ -27,18 +27,14 @@ local mirrorMaster = function (_, existing, new)
   end
 end
 
-local function oddPage ()
-  return tp == "odd"
+function package.oddPage (_)
+  return _odd
 end
 
 local function switchPage (class)
-  if class:oddPage() then
-    tp = "even"
-    class:switchMaster(class.evenPageMaster)
-  else
-    tp = "odd"
-    class:switchMaster(class.oddPageMaster)
-  end
+  _odd = not class:oddPage()
+  local nextmaster = class[(_odd and "odd" or "even") .. "PageMaster"]
```

Comment:
`local nextmaster = _odd and class.oddPageMaster or class.evenPageMaster` could be more legible, or did I miss something?

Reply:
Roger that. I was trying to do something else here but when it didn't work I forgot to go back and simplify again.